packages/@ember/-internals/glimmer/lib/utils/iterable.ts

```diff
@@ -127,7 +135,22 @@ class ObjectIterator extends BoundedIterator {
     } else {
       let values: Opaque[] = [];
       for (let i = 0; i < length; i++) {
-        values.push(get(obj, keys[i]));
+        let value: any;
+        let key = keys[i];
+
+        value = obj[key];
+
+        // Add the tag of the returned value if it is an array, since arrays
+        // should always cause updates if they are consumed and then changed
```

Comment:
Is that right? If you pass this to, e.g. `#each`, we handle this there already. If you just pass the value to, say, a component or a helper, I am not sure we need to invalidate those if the content of the array changes?

Reply:
I think the test you added should pass without consuming `[]`.

Reply:
This is part of autotracking as a whole, it allows consumers to use arrays (both native and non-native) anywhere they consume a tracked value, and have mutations to that array be autotracked. In practice, this is like the native getters migration, it’ll mean users can access any array as if it were a normal, native array. They only need to use methods for updating the state of the array (we need native Proxy to allow them to update with native syntax).

Even when we do get to native Proxy though, this will allow us to omit a `get` handler in the Proxy, which should help us avoid perf issues with proxies